src/OrchardCore.Modules/OrchardCore.ContentFields/Services/DefaultContentPickerResultProvider.cs

```diff
@@ -0,0 +1,44 @@
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Records;
+using YesSql;
+using YesSql.Services;
+
+namespace OrchardCore.ContentFields.Services
+{
+    public class DefaultContentPickerResultProvider : IContentPickerResultProvider
+    {
+        private readonly IContentManager _contentManager;
+        private readonly ISession _session;
+
+        public DefaultContentPickerResultProvider(IContentManager contentManager, ISession session)
+        {
+            _contentManager = contentManager;
+            _session = session;
+        }
+
+        public string Name => "Default";
+
+        public async Task<IEnumerable<ContentPickerResult>> Search(ContentPickerSearchContext searchContext)
```

Comment:
You can't return a list without limiting the number of items, or paging. But I will have other comments that might make this point useless.

Reply:
I limited it to 20 results